model-optimizer/mo/ops/memoryoffset.py

```diff
@@ -58,9 +57,19 @@ def infer(node: Node):
             elif pair_node.in_port(0).get_source().node.has_valid('out-size'):
                 out_size = pair_node.in_port(0).get_source().node['out-size']
                 node.out_port(0).data.set_shape(np.array([1, out_size]))
-            elif pair_node.in_port(0).get_source().node.op in ["Add", "ReLU"] and \
-                    pair_node.in_port(0).get_source().node.in_port(0).get_source().node.has_valid('out-size'):
-                out_size = pair_node.in_port(0).get_source().node.in_port(0).get_source().node['out-size']
+            elif pair_node.in_port(0).get_source().node.has_valid('shape'):
+                out_shape = pair_node.in_port(0).get_source().node['shape']
+                node.out_port(0).data.set_shape(out_shape)
+            elif pair_node.in_port(0).get_source().node.op in ["Add", "ReLU"]:
+                if pair_node.in_port(0).get_source().node.in_port(0).get_source().node.has_valid('out-size'):
+                    out_size = pair_node.in_port(0).get_source().node.in_port(0).get_source().node['out-size']
+                    node.out_port(0).data.set_shape(np.array([1, out_size]))
+                elif pair_node.in_port(0).get_source().node.in_port(1).get_source().node.has_valid('out-size'):
+                    out_size = pair_node.in_port(0).get_source().node.in_port(1).get_source().node['out-size']
+                    node.out_port(0).data.set_shape(np.array([1, out_size]))
+            elif pair_node.in_port(0).get_source().node.op == "Mul" and \
+                    pair_node.in_port(0).get_source().node.in_port(0).get_source().node.has_valid('shape'):
+                out_size = pair_node.in_port(0).get_source().node.in_port(0).get_source().node.shape
```

Comment:
It looks ugly but it works

Reply:
Definitely need some comments here describing what is going on.

Reply:
Let's save `pair_node.in_port(0).get_source().node` to a dedicated variable with a meaningful name. It will make a code much more readable. Also, we may introduce additional variables and refactor the whole `else` block starting from line 50

Reply:
Agree, I was doing this at the moment